src/frontend/src/pages/Login/Login.tsx

```diff
@@ -1,9 +1,27 @@
 import { Auth } from '@supabase/auth-ui-react';
 import { ThemeSupa } from '@supabase/auth-ui-shared';
+import useAuth from 'hooks/AuthHooks';
 import { supabase } from 'main';
+import AppRoutes from 'navigation/AppRoutes';
+import { Navigate } from 'react-router-dom';
 
 function Login() {
-    return <Auth supabaseClient={supabase} appearance={{ theme: ThemeSupa }} providers={[]}></Auth>;
+    const { session, isLoading } = useAuth();
+    // I want auto redirect but it's not working
+    if (!session) {
+        return (
+            <div className='loginClass'>
+                <Auth
+                    supabaseClient={supabase}
+                    appearance={{ theme: ThemeSupa }}
+                    providers={[]}
+                    redirectTo={`${AppRoutes.root}`}
+                ></Auth>
+            </div>
+        );
+    } else {
+        return <Navigate replace to='/' />;
```

Comment:
the `redirect` hook method is what you're looking for here. This Navigate element is a wrapper on the `navigate` hook method and not what we're looking for here,

See here for an example: https://reactrouter.com/en/main/fetch/redirect